Zend/zend_language_scanner.l

```diff
@@ -1245,11 +1260,11 @@ restart:
 
 /*!re2c
 re2c:yyfill:check = 0;
-LNUM	[0-9]+
-DNUM	([0-9]*"."[0-9]+)|([0-9]+"."[0-9]*)
+LNUM	[0-9]+(_[0-9]+)*
+DNUM	(([0-9]+(_[0-9]+)*)*"."([0-9]+(_[0-9]+)*)+)|(([0-9]+(_[0-9]+)*)+"."([0-9]+(_[0-9]+)*)*)
```

Comment:
```
(
    ([0-9]+(_[0-9]+)*)* // Writing the outer ()* as ()? would be clearer
    "."
    ([0-9]+(_[0-9]+)*)+ // The outer ()+ seems redundant?
)|...
```

I think this would be easier to understand if you reused `LNUM`, something like:

```
({LNUM}?"."{LNUM})|({LNUM}"."{LNUM}?)
```

Reply:
That does indeed make it easier to understand. Fixed.